### ISSUE TITLE
GET Query error

### ISSUE THREAD

**Issue by Dal-Papa** [PredictionIO/template-scala-parallel-universal-recommendation#37]
Hello !!

After struggling a lot with the default AWS install for couple hours and giving up, taking an empty EC2 node and use the quick install method, I finally was able to train my model correctly using the Universal Recommender template.

Unfortunately, when I query the engine when it's running, I get this : 

```
Query:
{ "user": "2abd37bc-67bf-4198-8139-f20b7db62c1a", "num": 4 }

Stack Trace:
org.elasticsearch.action.search.SearchPhaseExecutionException: Failed to execute phase [query], all shards failed; shardFailures
```

with a very long stack trace about Elastic (that I can copy if needed).

What exactly did I do wrong here ?
I get the exact same thing when querying with empty parameters. 
Happy to provide any required information.

Thanks in advance.

**Comment by goliasz** [comment]
BTW what is you UR event model? You can try my preconfigured docker container with UR engine. It is available here https://hub.docker.com/r/goliasz/raas-micro/. You can pull it and use directly (just fake PubNub keys) or you can train and query it via PubNub queues. If you wish I can spin-up the container for you.

**Comment by Dal-Papa** [comment]
The typical event looks like this : 

```
{"event":"open","entityType":"user","entityId":"C9A951BD-7D65-48BE-B823-838E97893093","targetEntityType":"item","targetEntityId":"television","properties":{"location":"xxxx,xxxx","language":"en-US"},"eventTime":"2016-02-21T12:01:59-05:00"}
```

and the engine conf : 

```
"algorithms": [
    {
      "comment": "simplest setup where all values are default, popularity based backfill, must add eventsNames",
      "name": "ur",
      "params": {
        "appName": "BlippApp",
        "indexName": "urindex",
        "typeName": "item",
        "comment": "must have data for the first event or the model will not build, other events are optional",
        "eventNames": ["open"]
      }
    }
  ]
```

Finally, I used `pio import --input some_json_file.json` in order to have events before the training. The events are there since I could get them with GET localhost:7070/events.json.

Thanks.

**Comment by goliasz** [comment]
Make sure you have corresponding event in "datasource" section of engine.json.

**Comment by Dal-Papa** [comment]
Here is the datasource : 

```
"datasource": {
    "params" : {
      "appName": "BlippApp",
      "eventNames": ["open"]
    }
  },
```

**Comment by Dal-Papa** [comment]
@goliasz : I installed your Docker container on the machine and it works ... How's that ?! Thanks anyway !!
@PredictionIOPeople : do you guys plan on having a full Docker deployment script or bundle ? Or if you have it, do you have a documentation for it ?

**Comment by goliasz** [comment]
@Dal-Papa : I'm not sure what may be the reason. Can you paste pio.log with errors?

**Comment by pferrel** [comment]
Sorry, we missed all these questions, if you could post to the Google group.

The primary problem is probably that until v0.3.0 the `"typeName"` in engine.json must be `"items"`. This is a bug fixed in v0.3.0, which is about to go to RC but you can get it from the repo v0.3.0 branch here is you want to have arbitrary URI compatible typeNames. https://github.com/actionml/template-scala-parallel-universal-recommendation/tree/v0.3.0

Another problem is that the event above is not of the right format. The event is a user event so the properties belongs to the user. You need to use a $set event to add properties to items as described here https://github.com/actionml/template-scala-parallel-universal-recommendation#property-change-events
